### PR TITLE
Implements new indicator - KST - KnowSureThing

### LINKED ISSUE #8126: Implenent Know Sure Thing
#### Expected Behavior
Supports Know Sure Thing (see [talipp implementation](https://github.com/nardew/talipp/blob/main/talipp/indicators/KST.py)).

Refer to [LEAN Engine > Contributions > Indicators](https://www.quantconnect.com/docs/v2/lean-engine/contributions/indicators).

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by femtotrader**
#### Related Issue

Closes #8126

#### Motivation and Context

This indicator wasn't implemented in QuantConnect Lean

#### Requires Documentation Change

Docstrings are written but there is probably room for better documentation

#### How Has This Been Tested?

Comparison with talipp computation of `KnowSureThing(5, 5, 10, 5, 15, 5 ,25, 10, 9, MovingAverageType.Simple)` of SPY candles using [generate_reference_data_from_talipp.py](https://github.com/QuantConnect/Lean/blob/master/Tests/TestData/generate_reference_data_from_talipp.py)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by femtotrader** [comment]
#8200 need to be fixed first. So converting this PR as draft

File: Algorithm/QCAlgorithm.Indicators.cs [hunk 1]
```diff
@@ -1114,6 +1114,38 @@ public KeltnerChannels KCH(Symbol symbol, int period, decimal k, MovingAverageTy
...
+        /// <param name="roc4MaPeriod">The smoothing period used to smooth the computed ROC4 values</param>
+        /// <param name="signalPeriod">The smoothing period used to smooth the signal values</param>
+        /// <param name="movingAverageType">Specifies the type of moving average to be used as smoother for KnowSureThing values</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to casting the input value to a TradeBar</param>
+        /// <returns>A new KnowSureThing indicator with the specified smoothing type and period</returns>
+        [DocumentationAttribute(Indicators)]
+        public KnowSureThing KST(Symbol symbol,
+            int roc1Period, int roc1MaPeriod, int roc2Period, int roc2MaPeriod,
+            int roc3Period, int roc3MaPeriod, int roc4Period, int roc4MaPeriod, int signalPeriod,
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.Indicators.cs:1138]: Think we could set some defaults values for the roc and ma periods? https://www.investopedia.com/terms/k/know-sure-thing-kst.asp

**Comment by Martin-Molinero** [comment]
Hey @femtotrader! Just checking in, would be nice to wrap this one up 👍

**Comment by Martin-Molinero** [comment]
Closing for now, please feel free to reopen once reviews are addressed 👍  thank you

**Pull request closed without merging**